enterprise/server/remote_execution/blockio/blockio.go

```diff
@@ -363,9 +364,25 @@ func (c *COWStore) ReadAt(p []byte, off int64) (int, error) {
 				p[i] = 0
 			}
 		} else {
-			if _, err := readFullAt(chunk, p[:readSize], chunkRelativeOffset); err != nil {
+			s, err := chunk.SizeBytes()
+			if err != nil {
 				return n, err
 			}
+			// Chunk might have less data available than the calculated size
+			// if this was the last chunk when we resized. If so then fill
+			// the range from [dataSize, readSize) with 0s.
+			dataSize := int64(readSize)
+			if remainder := s - chunkRelativeOffset; dataSize > remainder {
```

Comment:
nit: Change to `if readSize > remainder` - I find that more intuitive to understand